File: LC/python/Roman to Integer.py
class Solution:
    # @return an integer
    def romanToInt(self, s):
        #when left is less than right, right - left
        #else: right + left
        num = {'I':1,'V':5,'X':10,'L':50,'C':100,'D':500,'M':1000}
        sum = 0
        last = 0
        for i in range(len(s)-1,-1,-1):
            if last > num[s[i]]:
                sum = sum-num[s[i]]
            else:
                sum = sum+num[s[i]]
            last = num[s[i]]
        return sum
            
        

File: LC/java/Roman to Integer.java
public class Solution {
    public int romanToInt(String s) {
        int res;
        
        HashMap<Character, Integer> map = new HashMap<Character, Integer>();
        map.put('I',1);
        map.put('V',5);
        map.put('X',10);
        map.put('L',50);
        map.put('C',100);
        map.put('D',500);
        map.put('M',1000);
        
        res = map.get(s.charAt(s.length()-1));
        for (int t = s.length()-2; t >= 0; t--){
            if (map.get(s.charAt(t)) >= map.get(s.charAt(t+1))){
                res += map.get(s.charAt(t));
            }
            else{
                res -= map.get(s.charAt(t));
            }
        }
        return res;
    }
}